go/client/wallet_api_listen_display.go

```diff
@@ -47,32 +48,53 @@ func newWalletNotificationDisplay(g *libkb.GlobalContext) *walletNotificationDis
 		baseNotificationDisplay: newBaseNotificationDisplay(g),
 		cli:                     cli,
 		deduper:                 make(map[string]bool),
+		useV1:                   useV1,
 	}
 }
 
-func deduperKey(details stellar1.PaymentDetailsLocal) string {
+func deduperCLIKey(details stellar1.PaymentCLILocal) string {
+	return fmt.Sprintf("%s:%s", details.TxID, details.Status)
+}
+
+func deduperFrontendKey(details stellar1.PaymentDetailsLocal) string {
 	return fmt.Sprintf("%s:%s", details.Summary.TxID, details.Summary.StatusSimplified)
 }
 
 func (d *walletNotificationDisplay) displayPaymentDetails(ctx context.Context, source string,
 	accountID stellar1.AccountID, paymentID stellar1.PaymentID) error {
 	notif := newWalletNotification(source)
-	details, err := d.cli.GetPaymentDetailsLocal(ctx, stellar1.GetPaymentDetailsLocalArg{
-		AccountID: accountID,
-		Id:        paymentID,
-	})
-	if err != nil {
-		errStr := err.Error()
-		notif.Error = &errStr
-		d.printJSON(notif)
-		return nil
-	}
-	if _, dupeMsg := d.deduper[deduperKey(details)]; dupeMsg {
-		return nil
+
+	if d.useV1 {
```

Comment:
Not sure if there's a cleaner way to do this, as there's a lot of duplication between the two paths.